### PR TITLE
Issue #71 Linked user_account to organization

### PR DESCRIPTION
**Pull request by RobertBeaudenon**
Features implemented:
- A user can now be linked to an existing organization (but it's optional to handle the case of sys admins)
- Displaying in the toolbar the username of logged in user and his role as well as his organization if he is in one.
- When an SA logins then he is redirected to manage organization component
- When another user logins he is redirected to the dashboard
- Made sure to delete the user info stored in a session locally when the user logout to keep it secure
- Updated Registration of user form to include a dropdown menu to choose an organization (again which is optional)

### CODE REVIEW

**Review by NicholasNagy — COMMENTED** [review]
I left some comments questioning certain things.

Everything is good, thanks for the added tests.

I will approve and merge once there is consensus regarding my reviewed comments.

File: client/src/app/components/signup/signup.component.ts [hunk 1]
```diff
@@ -14,6 +15,8 @@ export class SignupComponent implements OnInit {
   signupForm: FormGroup;
   errorMessage: string;
   body: any;
+  organizations = [];
+  selectedOrganization;
```
[the remark this answers is not in the document]
Comment (RobertBeaudenon) [reply client/src/app/components/signup/signup.component.ts:19]: what do you mean by typing?

[the remark this answers is not in the document]
Comment (NicholasNagy) [reply client/src/app/components/signup/signup.component.ts:19]: Defining types for your variables.

File: client/src/app/components/login/login.component.ts [hunk 2]
```diff
@@ -42,10 +42,18 @@ export class LoginComponent implements OnInit {
       (data) => {
         this.tokenService.SetToken(data.token); // Setting token in cookie for logged in users
         if (data.role === 'SA') {
+          localStorage.setItem('role', data.role);
+          localStorage.setItem('username', data.user);
```
[the remark this answers is not in the document]
Comment (RobertBeaudenon) [reply client/src/app/components/login/login.component.ts:47]: I added the comments

**Review by NicholasNagy — APPROVED** [review]
Everything looks good now, thanks for the updates

**Pull request merged**